pkg/xen-tools/patches-4.19.0-rc4-v3/11-char-socket-revert.patch

```diff
@@ -0,0 +1,24 @@
+diff --git a/tools/qemu-xen/chardev/char-socket.c b/tools/qemu-xen/chardev/char-socket.c
+index ef62dbf..7ad180e 100644
+--- a/tools/qemu-xen/chardev/char-socket.c
++++ b/tools/qemu-xen/chardev/char-socket.c
+@@ -177,16 +177,15 @@ static int tcp_chr_write(Chardev *chr, const uint8_t *buf, int len)
+ 
+         if (ret < 0 && errno != EAGAIN) {
+             if (tcp_chr_read_poll(chr) <= 0) {
+-                /* Perform disconnect and return error. */
+                 tcp_chr_disconnect_locked(chr);
++                return len;
+             } /* else let the read handler finish it properly */
+         }
+ 
+         return ret;
+     } else {
+-        /* Indicate an error. */
```

Comment:
If you look to the whole function, this patch is basically ignoring any error, do we really want to do that? I think we should take this opportunity to review all these patches and understand when/why there are still need it or not. And this specific patch shows why it's important to add a good description to the patch, like here: https://github.com/lf-edge/eve/blob/master/pkg/xen-tools/patches-4.15.0/13-qemu-Add-cpu-pin-and-cpumask-options.patch I think most of these patches were introduced long time ago, and now it can be trick to recall the reason for each one of them....

Reply:
I have removed the patch section that skips the error.